examples/BERT/model.py

```diff
@@ -171,21 +176,137 @@ class CrossLingualMLMTask(nn.Module):
 
     def __init__(self, ntoken, ninp, nhead, nhid, nlayers, dropout=0.5):
         super(CrossLingualMLMTask, self).__init__()
-        self.xlmr_model = XLMRModel(ntoken, ninp, nhead, nhid, nlayers, dropout=dropout)
+
+        # self.xlmr_model = XLMRModel(ntoken, ninp, nhead, nhid, nlayers, dropout=dropout)
+        self.xlmr_embed = XLMREmbedding(ntoken, ninp, dropout)
+        encoder_layers = TransformerEncoderLayer(ninp, nhead, nhid, dropout)
+        self.transformer_encoder1 = TransformerEncoder(encoder_layers, nlayers // 2)
+        # ======================== Split ==========================
+        self.transformer_encoder2 = TransformerEncoder(encoder_layers, nlayers // 2)
+
         self.mlm_span = Linear(ninp, ninp)
         self.activation = F.gelu
         self.norm_layer = LayerNorm(ninp, eps=1e-12)
         self.mlm_head = Linear(ninp, ntoken)
 
     def forward(self, src):
-        output = self.xlmr_model(src)
+        # output = self.xlmr_model(src)
+        output = self.xlmr_embed(src)
+        output = self.transformer_encoder1(output)
+        # ======================== Split ==========================
+        output = self.transformer_encoder2(output)
+
         output = self.mlm_span(output)
         output = self.activation(output)
         output = self.norm_layer(output)
         output = self.mlm_head(output)
         return output
 
 
+def get_cuda_if_available(i):
+    assert i >= 0
+    if torch.cuda.is_available():
+        return f"cuda:{min(i, torch.cuda.device_count() - 1)}"
+    else:
+        return "cpu"
+
+
+class CrossLingualMLMTaskBase(nn.Module):
```

Comment:
This module seems not task specific. Maybe we can put it in pytorch distributed later on.